vsintegration/src/FSharp.Editor/LanguageService/LanguageService.fs

```diff
@@ -89,6 +92,12 @@ type internal FSharpWorkspaceServiceFactory
                 | _ ->
                     None
 
+            let getSource filename =
+                workspace.CurrentSolution.TryGetDocumentFromPath(filename)
+                |> Option.map(fun document ->
+                    let text = document.GetTextAsync().Result
```

Comment:
Any safer way to do this? Accessing the result directly can backfire sometimes.

Reply:
Enabling this to be async all the way through would be quite complicated unfortunately. 

So barring that we have to block here. And I don't know if there is any better way to do that than using `.Result`.

Reply:
Hmm how about the sync thing `document.TryGetText`?

Reply:
That's what it already does: https://sourceroslyn.io/#Microsoft.CodeAnalysis.Workspaces/Workspace/Solution/TextDocumentState.cs,138

However that only returns if it's already been loaded. 

Also we already use `document.GetTextAsync().Result` in a few other places. 

Definitely not ideal, but to do it properly will probably take significant refactoring of IncrementalBuilder.

Reply:
> Hmm how about the sync thing `document.TryGetText`? 

It'll only check if text is fetched already and return if it is, and won't perform IO if it's not.

Reply:
I see. Well I can live with this although not as happily as I used to.
Probably could be a good first issue to asynchronize these flows properly.